File: Lunchtime/Controller/SetupPageSecondViewController.m
//
//  SetupPageThirdViewController.m
//  Lunchtime
//
//  Created by Willow Belle on 2015-11-13.
//  Copyright © 2015 Cosmic Labs. All rights reserved.
//

#import "SetupPageSecondViewController.h"
#import "User.h"

@interface SetupPageSecondViewController ()

@property (nonatomic, weak) IBOutlet UISegmentedControl *priceLimitSegmentedControl;

@end

@implementation SetupPageSecondViewController

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];

    RLMRealm *realm = [RLMRealm defaultRealm];
    User *user = [User new];

    [user setPriceLimit:(unsigned)self.priceLimitSegmentedControl.selectedSegmentIndex];
    [user setIdentifier:@1];

    [realm beginWriteTransaction];
    [User createOrUpdateInRealm:realm withValue:user];
    [realm commitWriteTransaction];
}

@end

File: Lunchtime/Controller/SetupPageSecondViewController.swift
//
//  SetupPageSecondViewController.swift
//  Lunchtime
//
//  Created by Willow Bumby on 2015-11-18.
//  Copyright © 2015 Lighthouse Labs. All rights reserved.
//

import UIKit
import Realm

class SetupPageSecondViewController: UIViewController {

    // MARK: - Outlets
    @IBOutlet weak var priceLimitSegmentedControl: UISegmentedControl?

    // MARK: - Controller Lifecycle
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)

        guard let selectedSegmentIndex = priceLimitSegmentedControl?.selectedSegmentIndex else {
            return
        }

        let realm = RLMRealm.defaultRealm()
        let user = User()

        user.priceLimit = selectedSegmentIndex
        user.identifier = NSNumber(integer: 1)

        do {
            realm.beginWriteTransaction()
            User.createOrUpdateInRealm(realm, withValue: user)
            try realm.commitWriteTransaction()
        } catch let error {
            print("Something went wrong...")
            print("Error Reference: \(error)")
        }
    }
}
